leenlab2/reminiscence
# Refactor puzzle logic to updated game flow

Refactor puzzle logic to be in line with our updated game play.



# Description

Refactor puzzle logic to prevent items from being interact-able when placed in the right spot, added two events for when branching item placed and non branching key item placed, ensure shadow cue for objects are no longer based on a timer, and fixed bug where player cannot insert tape if crosshair is not directly on TV Screen

### Type of change






Bug Fix
## How to Integrate


# How Has This Been Tested?

Open the Main scene and test the following:

- Before inserting the tape, check to make sure none of the cues are showing
- Insert the tape. The poster cue should show immediately (like when the user inserts the tape and enters into the memory world, the poster cue will show immediately)
- Place one poster in the correct location on the wall. 
-        Confirm you can no longer pick up that poster 
-        Confirm you can still pick up the other poster but cannot put this poster anywhere on the walls. You should still be able to put this poster on any flat surface.

- Note the key items’ shadow cues won’t turn on yet since we still need to “exit the memory and come back in”. For now I’ve added a function called ShowNonBranchingItemsShadowCues() to be called when we have the “exit memory” functionality

- Place the other key items for this branch down. Ensure the key items placed in the right spot cannot be pick-upable again after.

- Try placing the key items of the other branch down. Confirm nothing happens and you can still pick it up again.

- Complete one branch and check TV still plays right solution
- Also pick up the tape and make sure you can insert it into the TV when the crosshair is on ANY part of the TV.




# Checklists
## Integration Checklist

## Review Checklist
